backend/corpora/lambdas/api/v1/collection.py

```diff
@@ -1,18 +1,34 @@
 from flask import make_response, jsonify
+from typing import Optional
 
-from ....common.corpora_orm import CollectionVisibility
+from ....common.corpora_orm import DbCollection, CollectionVisibility
 from ....common.utils.db_utils import db_session
 from ....common.entities import Collection
 from ....common.utils.exceptions import ForbiddenHTTPException
 
 
 @db_session
-def get_collections_list(user_uuid: str = "", from_date: int = None, to_date: int = None, user=None):
-    result = dict(collections=Collection.list_collections_in_time_range(from_date=from_date, to_date=to_date))
+def get_collections_list(from_date: int = None, to_date: int = None, user: Optional[str] = None):
+
+    all_collections = Collection.list_attributes_in_time_range(
+        from_date=from_date,
+        to_date=to_date,
+        list_attributes=[DbCollection.id, DbCollection.visibility, DbCollection.owner, DbCollection.created_at],
+    )
+
+    collections = []
+    for coll_dict in all_collections:
+        visibility = coll_dict["visibility"]
+        owner = coll_dict["owner"]
```

Comment:
this might be a premature optimization, but wouldn't it be better to write a query that filters by owner or public rather than returning all of the collections within a time range. I think @mckinsel could help write that query.

Reply:
I thought about do that, but then I figured that when the get to the point where there are a large number of collections, then ideally the number of public collections should by much greater than the number of private, so the optimization wouldn't save too much.